src/main/java/org/opentripplanner/graph_builder/module/osm/WalkableAreaBuilder.java

```diff
@@ -314,13 +307,34 @@ public void buildWithVisibility(AreaGroup group) {
 
       createNamedAreas(edgeList, ring, group.areas);
 
+      if (visibilityNodes.size() > maxAreaNodes) {
+        issueStore.add(new AreaTooComplicated(group, visibilityNodes.size(), maxAreaNodes));
+      }
+
+      // if area is too complex, consider only part of visibility nodes
+      // so that at least some edges passing through the area is added
+      // otherwise routing can use only area boundary edges
+      float skip_ratio = (float) maxAreaNodes / (float) visibilityNodes.size();
+      int i = 0;
+      float sum_i = 0;
       for (OSMNode nodeI : visibilityNodes) {
+        sum_i += skip_ratio;
```

Comment:
Nice fix!

Just a note that summing floats like this can actually be a bit dangerous. If skip_ratio is sufficiently small (like 1e-8) the sum will never reach 1. It probably won't be issue in this case but you might want to write this logic using an integer counter instead

Reply:
Right. In this case, float increment is typically greater than 0.1, and in some truly exceptional cases > 0.01. Lower values would mean that street graph is useless for routing, and it won't matter if area edges will not get added.